### DISCUSSION TITLE
Motif enrichment in a subset of peaks

### DISCUSSION THREAD

**Discussion by aselewa** [GreenleafLab/ArchR#880] [Questions / Documentation]
Hi,

I am wondering if theres a way to estimate motif enrichment in any subset of the union peak set? In other words, I am trying to estimate motif enrichment in peaks that are shared across all cell-types. I was able to define the shared peaks using my own way, but I am wondering how I can feed these to peakAnnoEnrichment?

Thanks so much.

**Comment by rcorces** [comment]
Just change your peakSet using `addPeakSet()`?

**Reply by aselewa** [reply]
As I understand, addPeakSet is for adding a set of features outside of ArchR. I would still like to use the union set constructed by ArchR.  
It seems currently that peakAnnoEnrichment only works on seMarker object from differential accessibility testing. Is it possible to run peakAnnoEnrichment on any subset of peaks from the union set? I think I got it to work in a hacky way, but wondering what others thought.

Thanks again.

**Reply by cristanchoa** [reply]
I have a similar question. I am trying to obtain the motif analysis on peaks shared across cell types but still from the same peaks that were initially called. Would you mind sharing how you think you got it to work, @aselewa?

**Reply by aselewa** [reply]
Hi, I dont access to my code anymore, but I remember formatting my union set to look like the seMarker object and giving that to peakAnnoEnrichment. You can print out the seMarker object from differential accessibility to see what it looks like, but its basically a summerized experiment object. Theres a few columns in the metadata but I left them as null and ArchR didnt complain.

Its pretty hacky so I would check if ArchR devs have made something since then.

**Reply by Y-ogativity** [reply]
I have a similar question. It would be very convenient for me to use the peakAnnoErichmet function with a custom set of peaks (e.g., shared peaks across cell types) that have already been called.

**Reply by trinevd** [reply]
Hi. I have a similar problem. I'd like to perform motif enrichment analysis on subset of peaks shared or unique across cell types, and get the same error.

**Reply by rcorces** [reply]
There is a new function on the `dev` branch called `customEnrichment()` which should enable what you are looking for.

https://github.com/GreenleafLab/ArchR/blob/c44b50c56323d5995fb3d2fb28a9072255fec644/R/AnnotationPeaks.R#L1053-L1146
